base/c.jl

```diff
@@ -43,6 +43,29 @@ end
 
 typealias Coff_t FileOffset
 
+# C NUL-terminated string pointers; these can be used in ccall
+# instead of Ptr{Cchar} and Ptr{Cwchar_t}, respectively, to enforce
+# a check for embedded NUL chars in the string (to avoid silent truncation).
+immutable Cstring; p::Ptr{Cchar}; end
+immutable Cwstring; p::Ptr{Cwchar_t}; end
```

Comment:
Okay, I have it working as a `bitstype` if I hard-code the size as 64.  I'm trying to figure out how to use `WORD_SIZE`, but it is tricky because a lot of things aren't defined yet at this point in the bootstrap process.